src/figuresClasses.ts

```diff
@@ -1,11 +1,77 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape = 'triangle',
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side cannot be less or equal to 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(
+        'The longest side cannot be more or equal to sum of the rest sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const semiperimeter: number = (a + b + c) / 2;
+    const area: number = Math.sqrt(
+      semiperimeter *
+        (semiperimeter - a) *
+        (semiperimeter - b) *
+        (semiperimeter - c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    public shape = 'circle',
```

Comment:
[CODE KNOWLEDGE] - In the Circle constructor, 'shape' is being assigned manually to 'this.shape'. Since you are using the public access modifier, TypeScript takes care of this automatically. You can remove the manual assignment.